### ISSUE TITLE
storage: emit rangefeed event upon lease transfer

### ISSUE THREAD

**Issue by ajwerner** [cockroachdb/cockroach#32979]
This issue relates to the work on closed timestamps and CDC for 2.2. Currently, when a lease transfer occurs and then no further activity on the range occurs, there will be no event emitted. The task is to eliminate this behavior by always emitting an event from the new lease holder upon lease transfer.

**Comment by ajwerner** [comment]
@tbg or @nvanbenschoten I created this issue based on my understanding of the problem after our discussion last week. After some digging today, it seems to me that there is a call upon lease transfer to EmitMLIA which is roughly what I expected this change to entail. See [here](https://github.com/cockroachdb/cockroach/blob/65bcb9b5f25a18a514cc067ed70cf516090873ba/pkg/storage/replica_proposal.go#L348-L351):
```
		// Emit an MLAI on the leaseholder replica, as follower will be looking
		// for one and if we went on to quiesce, they wouldn't necessarily get
		// one otherwise (unless they ask for it, which adds latency).
		r.EmitMLAI()
```

Can either of y'all help provide more context about what you have in mind for the change?

**Comment by tbg** [comment]
Hmm, I'm blanking on what the problem was if not that. @nvanbenschoten surely remembers, though.

**Comment by nvanbenschoten** [comment]
I also thought that was the problem, but the code looks fine. It's possible that there's a variant of a lease transfer that doesn't kick off closed timestamp emission.

We saw this pretty clearly when running the CDC roachtests with Rangefeeds enabled. It would be worth it to sync up with @danhhz about running those and then to track down the cause of stalled resolved timestamps, which was due to this kind of problem.
